src/test/java/io/vertx/kafka/client/tests/ConsumerTestBase.java

```diff
@@ -547,6 +552,162 @@ public void testPositionNonEmptyTopic(TestContext ctx) throws Exception {
     consumer.subscribe(Collections.singleton("the_topic"));
   }
 
+
+
+  /*
+    Tests beginningOffset
+   */
+  @Test
+  public void testBeginningOffset(TestContext ctx) throws Exception {
+   testBeginningEndOffset(ctx, true);
+  }
+
+  /*
+    Teste endOffset (boolean parameter = false)
+   */
+  @Test
+  public void testEndOffset(TestContext ctx) throws Exception {
+    testBeginningEndOffset(ctx, false);
+  }
+
+  /*
+   Tests test beginningOffset or endOffset, depending on beginningOffset = true or false
+  */
+  public void testBeginningEndOffset(TestContext ctx, boolean beginningOffset) throws Exception {
+    KafkaCluster kafkaCluster = kafkaCluster().addBrokers(1).startup();
+    Async batch = ctx.async();
+    AtomicInteger index = new AtomicInteger();
+    int numMessages = 1000;
+    kafkaCluster.useTo().produceStrings(numMessages, batch::complete, () ->
+      new ProducerRecord<>("the_topic", 0, "key-" + index.get(), "value-" + index.getAndIncrement()));
+    batch.awaitSuccess(20000);
+
+    Properties config = kafkaCluster.useTo().getConsumerProperties("the_consumer", "the_consumer", OffsetResetStrategy.EARLIEST);
+    config.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+    config.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+
+    Context context = vertx.getOrCreateContext();
+    consumer = createConsumer(context, config);
+
+    consumer.exceptionHandler(ctx::fail);
+
+    Set<TopicPartition> topicPartitions = new HashSet<>();
+    TopicPartition topicPartition= new TopicPartition("the_topic", 0);
+    topicPartitions.add(topicPartition);
+
+    // Test contains three sub-tests, so we need to countdown three times for completion
+    Async done = ctx.async(2);
+    consumer.handler(handler -> {
+      // nothing to do in this test
+    });
+
+    consumer.subscribe(Collections.singleton("the_topic"), subscribeRes -> {
+        if (subscribeRes.succeeded()) {
+          if(beginningOffset) {
+            consumer.beginningOffsets(topicPartitions, beginningOffsetResult -> {
```

Comment:
The first one tests the method which gets a Set as input, the second one tests the "convenience" method with only one TopicPartition as input. The second one does return a Long only and no Set/Collection, such no size check